test/scripts/upgrade-ovn.sh

```diff
@@ -0,0 +1,136 @@
+    READY_REPLICAS=$(run_kubectl get daemonsets.apps $1 -n ovn-kubernetes -o=jsonpath='{.status.numberReady}')
+    echo "CURRENT READY REPLICAS: $READY_REPLICAS, CURRENT DESIRED REPLICAS: $DESIRED_REPLICAS for the DaemonSet $1"
+    if [[ $READY_REPLICAS -eq $DESIRED_REPLICAS ]]; then
+      UP_TO_DATE_REPLICAS=$(run_kubectl get daemonsets.apps ovnkube-node -n ovn-kubernetes  -o=jsonpath='{.status.updatedNumberScheduled}')
+      echo "CURRENT UP TO DATE REPLICAS: $UP_TO_DATE_REPLICAS for the Deployment $1"
+      if [[ $READY_REPLICAS -eq $UP_TO_DATE_REPLICAS ]]; then
+        break
+      fi
+    fi
+    sleep 30
+  done
+  
+}
+
+kubectl_wait_deployment(){
+  # takes one deployment and makes sure its replicas and readyReplicas are equal
+  while true; do
+    run_kubectl get deployments.apps $1 -n ovn-kubernetes 
+    DESIRED_REPLICAS=$(run_kubectl get deployments.apps $1 -n ovn-kubernetes -o=jsonpath='{.status.replicas}')
+    READY_REPLICAS=$(run_kubectl get deployments.apps $1 -n ovn-kubernetes -o=jsonpath='{.status.readyReplicas}')
+    echo "CURRENT READY REPLICAS: $READY_REPLICAS, CURRENT DESIRED REPLICAS: $DESIRED_REPLICAS for the Deployment $1"
+    if [[ $READY_REPLICAS -eq $DESIRED_REPLICAS ]]; then
+      UP_TO_DATE_REPLICAS=$(run_kubectl get deployments.apps ovnkube-master -n ovn-kubernetes -o=jsonpath='{.status.updatedReplicas}')
+      echo "CURRENT UP TO DATE REPLICAS: $UP_TO_DATE_REPLICAS for the Deployment $1"
+      if [[ $READY_REPLICAS -eq $UP_TO_DATE_REPLICAS ]]; then
+        break
+      fi
+    fi
+    sleep 30
+  done
+  
+}
+
+run_kubectl() {
+  local retries=0
+  local attempts=10
+  while true; do
+    if kubectl "$@"; then
+      break
+    fi
+
+    ((retries += 1))
+    if [[ "${retries}" -gt ${attempts} ]]; then
+      echo "error: 'kubectl $*' did not succeed, failing"
+      exit 1
+    fi
+    echo "info: waiting for 'kubectl $*' to succeed..."
+    sleep 1
+  done
+}
+
+
+install_ovn_image() {
+  kind load docker-image "${OVN_IMAGE}" --name "${KIND_CLUSTER_NAME}"
+}
+
+kubectl_wait_for_upgrade(){
+    # waits until new image is updated into all relevant pods
+    count=0
+    while [ $count -lt 5 ];
+    do
+        echo "waiting for ovnkube-master, ovnkube-node, ovnkube-db to have new image, ${OVN_IMAGE}, sleeping 30 seconds"
+        sleep 30
+        count=$(run_kubectl get pods -n ovn-kubernetes -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}'|grep -c ${OVN_IMAGE})
+        echo "Currently count is $count, expected 5"
+    done;
+
+}
+
+## This script is responsible to upgrade ovn daemonsets to run new pods with image built from a PR
+install_ovn_image
+run_kubectl set image daemonsets.apps ovnkube-node ovnkube-node="${OVN_IMAGE}" ovs-metrics-exporter="${OVN_IMAGE}"  ovn-controller="${OVN_IMAGE}" -n ovn-kubernetes
+sleep 30
```

Comment:
do we need the sleeps (below too) if the next step is going to poll on what's important anyway?

Reply:
maybe some leftovers, let me remove and re-run shouldn't affect the outcome.